### PR TITLE
Data validation and error handling for external entities

### PR DESCRIPTION
**Pull request by johannahemminger**
https://mendix.atlassian.net/browse/TW-1643

### CODE REVIEW

**Comment by johannahemminger** [comment]
@ezagroba can you check this out?

**Comment by johannahemminger** [comment]
@joostverhoog Could you check this out? Thx

File: content/en/docs/refguide/modeling/integration/consumed-odata-services/consumed-odata-service-requirements.md [hunk 1]
```diff
@@ -133,3 +133,17 @@ For each call, the app instance logs how many objects it has retrieved and how m
 {{% alert color="warning" %}}
 It is up to the application to communicate to its end-users that the daily limit has been reached. If this is not done, the end-user gets a message that an error occurred.
 {{% /alert %}}
+
+## 5 Validating Data and Error Handling
+
+You can [set up data validation](/howto/data-models/setting-up-data-validation/) to check on inserted or updated data between your apps. An insert validation microflow in the domain model of the client application - the application consuming the service - can check on the data that is inserted or updated.
```
Comment (joostverhoog) [inline content/en/docs/refguide/modeling/integration/consumed-odata-services/consumed-odata-service-requirements.md]: The page consumed-odata-service-requirements is intended to list the requirements that a Mendix app has on the service that is being consumed. I suggest that we write about how to validate incoming updates, inserts and deletes in /refguide/published-odata-resource/, and write about how to handle those validation errors in in /refguide/send-external-object/ and /refguide/delete-external-object/.

File: content/en/docs/refguide/modeling/integration/consumed-odata-services/consumed-odata-service-requirements.md [hunk 2]
```diff
@@ -133,3 +133,17 @@ For each call, the app instance logs how many objects it has retrieved and how m
...
+
+How does the published service know that that an insert fails? In the publishing app, you can use a validation message action to report the error. The client app can include a custom error handler on the [Send External Object](/refguide/send-external-object/) activity to handle the error.
+
+The domain model for the publishing app's validation message would look like this:
+
+{{< figure src="/attachments/refguide/modeling/integration/consumed-odata-service-requirements/validate-data-publishing-app.png" alt="publishing app validation" >}}
+
+In the client app, the error handler would look like this:
+
+{{< figure src="/attachments/refguide/modeling/integration/consumed-odata-service-requirements/validate-data-client-app.png" alt="client app validation" >}}
```
Comment (joostverhoog) [inline content/en/docs/refguide/modeling/integration/consumed-odata-services/consumed-odata-service-requirements.md]: I think examples are better suited for the how-to section of the docs, not in the refguide. That's also the place where the relation between different features is better explained. Maybe we can turn this into a how-to?

**Review by joostverhoog — CHANGES_REQUESTED** [review]

File: content/en/docs/refguide/modeling/integration/consumed-odata-services/consumed-odata-service-requirements.md (hunk 1 above)
Comment (johannahemminger) [reply content/en/docs/refguide/modeling/integration/consumed-odata-services/consumed-odata-service-requirements.md]: Sounds good, I'll make those changes!

File: content/en/docs/refguide/modeling/integration/consumed-odata-services/consumed-odata-service-requirements.md (hunk 2 above)
Comment (johannahemminger) [reply content/en/docs/refguide/modeling/integration/consumed-odata-services/consumed-odata-service-requirements.md]: Sure, I'll create a task to do that. Thanks for your input!

**Pull request merged**